KraflaPartialLibrary.aml

```diff
@@ -0,0 +1,193 @@
+          <Description>Type of signal communcication</Description>
+          <DefaultValue />
+          <Value />
+          <Constraint Name="LegalValues">
+            <NominalScaledType>
+              <RequiredValue>Signal Interface</RequiredValue>
+              <RequiredValue>Bus Interface</RequiredValue>
+            </NominalScaledType>
+          </Constraint>
+        </Attribute>
+        <Attribute Name="Direction" AttributeDataType="xs:string">
+          <Description />
+          <DefaultValue />
+          <Value>In</Value>
+          <Constraint Name="LegalValues">
+            <NominalScaledType>
+              <RequiredValue>In</RequiredValue>
+            </NominalScaledType>
+          </Constraint>
+        </Attribute>
+        <Attribute Name="SignalClass" AttributeDataType="xs:string">
+          <Description />
+          <DefaultValue />
+          <Value>Binary</Value>
+          <Constraint Name="LegalValues">
+            <NominalScaledType>
+              <RequiredValue>Binary</RequiredValue>
+            </NominalScaledType>
+          </Constraint>
+        </Attribute>
+      </ExternalInterface>
+      <ExternalInterface Name="FlowIn" ID="596fc621-5370-4e70-adc7-b0c70b41b3a5" RefBaseClassPath="InterfaceClassLibrary/ContextHidingFlowClass/ContextHidingFlowIn">
+        <Attribute Name="Direction" AttributeDataType="xs:string">
+          <Value>In</Value>
+          <Constraint Name="LegalValues">
+            <NominalScaledType>
+              <RequiredValue>In</RequiredValue>
+            </NominalScaledType>
+          </Constraint>
+        </Attribute>
+      </ExternalInterface>
+      <ExternalInterface Name="FlowOut" ID="3a050005-6825-4c98-b57d-40d9893802c8" RefBaseClassPath="InterfaceClassLibrary/ContextHidingFlowClass/ContextHidingFlowOut">
+        <Attribute Name="Direction" AttributeDataType="xs:string">
+          <Value>Out</Value>
+          <Constraint Name="LegalValues">
+            <NominalScaledType>
+              <RequiredValue>Out</RequiredValue>
+            </NominalScaledType>
+          </Constraint>
+        </Attribute>
+      </ExternalInterface>
+    </SystemUnitClass>
+    <SystemUnitClass Name="Context" ID="fef18617-1f55-4f40-95f3-ba2eb8b81cce">
+      <Attribute Name="Description" AttributeDataType="xs:string" />
+      <Attribute Name="NoteRef" AttributeDataType="xs:string">
+        <Description>Reference to Drawing Note object GUID</Description>
+        <DefaultValue />
+        <Value />
+        <Constraint Name="aml-RegExp">
+          <UnknownType>
+            <Requirements>^[0-9A-Fa-f]{8}(-[0-9A-Fa-f]{4}){4}[0-9A-Fa-f]{8}$</Requirements>
+          </UnknownType>
+        </Constraint>
+      </Attribute>
+      <Attribute Name="TimeDelayOff" AttributeDataType="xs:duration" />
+      <Attribute Name="AlarmsToHide" AttributeDataType="">
+        <Attribute Name="AlarmToHide1" AttributeDataType="" />
```

Comment:
Does the AML standard allow nesting of attributes?

Reply:
Yes, nested attributes are allowed.
IEC 62714 describes modelling of lists:
![image](https://user-images.githubusercontent.com/89920801/157011422-f94e7de6-2917-4817-b52a-3837b898b6b1.png)

Structurally it is correct to put list elements under parent attribute. And since we have already switched to CAEX 3, base attribute type library (AutomationMLBaseAttributeTypeLib) should be added to library (to iec63131 library branch) and AlarmsToHide attribute should reference to ListType attribute type.

This requires changes on main iec63131 library and will be done in next revision.